twitterpuller/src/test/java/de/brandwatch/minianalytics/twitterpuller/twitter/TweetFetcherTest.java

```diff
@@ -0,0 +1,52 @@
+package de.brandwatch.minianalytics.twitterpuller.twitter;
+
+
+import de.brandwatch.minianalytics.twitterpuller.kafka.Producer;
+import de.brandwatch.minianalytics.twitterpuller.model.Resource;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import static org.junit.Assert.*;
+
+import static org.mockito.Mockito.*;
+
+import org.mockito.junit.MockitoJUnitRunner;
+import twitter4j.Status;
+import twitter4j.User;
+
+import java.util.Date;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class TweetFetcherTest {
+
+    @Mock
+    Producer producer;
+
+    @Test
+    public void testIfResourceObjectIsGeneratedCorrectly() {
+
+        Status status = mock(Status.class, RETURNS_DEEP_STUBS);
+
+        Date date = new Date();
+
+        when(status.getText()).thenReturn("Hello World");
+        when(status.getUser().getName()).thenReturn("Max Leopold");
+        when(status.getCreatedAt()).thenReturn(date);
+
+        ArgumentCaptor<Resource> argumentCaptor = ArgumentCaptor.forClass(Resource.class);
+
+        TwitterPullerStatusListener twitterPullerStatusListener = new TwitterPullerStatusListener(producer);
+        TweetFetcher tweetFetcher = new TweetFetcher(twitterPullerStatusListener);
+
+        twitterPullerStatusListener.onStatus(status);
+
+        verify(producer).send(argumentCaptor.capture());
+
+        assertEquals("Hello World", argumentCaptor.getValue().getText());
```

Comment:
This is one way to do it. You should alternatively also be able to create a `expectedResource` object in this test, and then just do one `verify(producer).send(eq(expectedResource));`

Reply:
This doesn't work. 
```
producer.send(
    de.brandwatch.minianalytics.twitterpuller.model.Resource@452e19ca
);
-> at de.brandwatch.minianalytics.twitterpuller.twitter.StatusListenerTest.testIfResourceObjectIsGeneratedCorrectly(StatusListenerTest.java:62)

However, there was exactly 1 interaction with this mock:
producer.send(
    de.brandwatch.minianalytics.twitterpuller.model.Resource@1ec9bd38
);
-> at de.brandwatch.minianalytics.twitterpuller.twitter.TwitterPullerStatusListener.onStatus(TwitterPullerStatusListener.java:35)
```
But i could do something pretty similar:

`assertTrue(new ReflectionEquals(expectedResource).matches(argumentCaptor.getValue()));`
This does work.

Reply:
What do you prefer?

Reply:
It will probably work if you implement `equals` and `hashCode` on your `Resource`? What IntelliJ generates for you is likely enough. It is currently comparing only the object hashes, which are obviously different. You want it to compare by the field values instead.

https://stackoverflow.com/questions/2265503/why-do-i-need-to-override-the-equals-and-hashcode-methods-in-java

Reply:
If I implement what IntelliJ generates, it just calls `super.equals(obj) `. And with that it still fails...

Reply:
Really? That's odd, if I do it it generates this:

```
@Override
public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Resource)) return false;
    Resource resource = (Resource) o;
    return Objects.equal(getAuthor(), resource.getAuthor()) &&
            Objects.equal(getText(), resource.getText()) &&
            Objects.equal(getDate(), resource.getDate());
}

@Override
public int hashCode() {
    return Objects.hashCode(getAuthor(), getText(), getDate());
}
```

You'll depend on [Guava](https://mvnrepository.com/artifact/com.google.guava/guava/28.1-jre) for that though:

Reply:
Oh, I see why. You can just type `equals` into your Code and IntelliJ generates an equals method like I have. But if you go cmd+m and then generate hashCode and equals you get the methods like you have... didn't knew they were different...

Reply:
With these hashCode/equal Methods `verify(producer).send(eq(expectedResource));` does work.